boat-scaffold/src/main/java/com/backbase/oss/codegen/java/BoatJavaCodeGen.java

```diff
@@ -1,16 +1,165 @@
+    public static final String USE_SET_FOR_UNIQUE_ITEMS = "useSetForUniqueItems";
+
+    public static final String USE_CLASS_LEVEL_BEAN_VALIDATION = "useClassLevelBeanValidation";
+    public static final String USE_JACKSON_CONVERSION = "useJacksonConversion";
+    public static final String SERVICE_ID = "serviceId";
+    public static final String CONFIG_NAME = "configName";
+
+    /**
+     * Whether to use {@code with} prefix for pojos modifiers.
+     */
+    @Setter
+    @Getter
+    protected boolean useWithModifiers;
+
+    @Setter
+    @Getter
+    protected boolean useSetForUniqueItems;
+
+    /**
+     * Add @Validated to class-level Api interfaces. Defaults to false
+     */
+    @Setter
+    @Getter
+    protected boolean useClassLevelBeanValidation;
+
+    /**
+     * Whether to use Jackson to convert query parameters to String.
+     */
+    @Setter
+    @Getter
+    protected boolean useJacksonConversion;
+
+    @Setter
+    @Getter
+    protected String serviceId;
+
+    @Setter
+    @Getter
+    protected String configName;
+
     public BoatJavaCodeGen() {
         this.embeddedTemplateDir = this.templateDir = NAME;
+
+        this.cliOptions.add(CliOption.newBoolean(USE_CLASS_LEVEL_BEAN_VALIDATION,
+            "Add @Validated to class-level Api interfaces", this.useClassLevelBeanValidation));
+        this.cliOptions.add(CliOption.newBoolean(USE_WITH_MODIFIERS,
+            "Whether to use \"with\" prefix for POJO modifiers", this.useWithModifiers));
+        this.cliOptions.add(CliOption.newBoolean(USE_SET_FOR_UNIQUE_ITEMS,
+            "Use java.util.Set for arrays that have uniqueItems set to true", this.useSetForUniqueItems));
+        this.cliOptions.add(CliOption.newBoolean(USE_JACKSON_CONVERSION,
+            "Whether to use Jackson to convert query parameters to String", this.useJacksonConversion));
+        this.cliOptions.add(CliOption.newString(SERVICE_ID,
+            "Generate the Backbase specific bean factory for ApiClient for the given serviceId"));
+        this.cliOptions.add(CliOption.newString(CONFIG_NAME,
+            "The configuration name under \"backbase.communication.services.\" - defaults to serviceId"));
     }
 
     @Override
     public String getName() {
         return NAME;
     }
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+
+        if (this.additionalProperties.containsKey(USE_WITH_MODIFIERS)) {
+            this.useWithModifiers = convertPropertyToBoolean(USE_WITH_MODIFIERS);
+        }
+        writePropertyBack(USE_WITH_MODIFIERS, this.useWithModifiers);
+
+        if (this.additionalProperties.containsKey(USE_SET_FOR_UNIQUE_ITEMS)) {
+            this.useSetForUniqueItems = convertPropertyToBoolean(USE_SET_FOR_UNIQUE_ITEMS);
+        }
+        writePropertyBack(USE_SET_FOR_UNIQUE_ITEMS, this.useSetForUniqueItems);
+
+        if (this.useSetForUniqueItems) {
+            this.typeMapping.put("set", "java.util.Set");
```

Comment:
Sonar will probably tell you to use contants here :)

Reply:
done